gui/src-tauri/src/main.rs

```diff
@@ -111,20 +140,30 @@ fn main() {
 	let stdout_recv = if let Some(p) = run_path {
 		log::info!("Server found on path: {}", p.to_str().unwrap());
 
-		let java_folder = p.join("jre");
-		let (recv, _child) =
-			Command::new(java_folder.join("bin/java").to_str().unwrap())
-				.current_dir(p)
-				.args(["-Xmx512M", "-jar", "slimevr.jar", "--no-gui"])
-				.spawn()
-				.expect("Unable to start the server jar");
+		// Check if any Java already installed is compatible
+		let java_paths = valid_java_paths();
+		let jre = p.join("jre/bin/java");
+		let java_bin = jre
+			.exists()
+			.then(|| jre.to_string_lossy())
```

Comment:
I'm on my phone so I wasn't able to test this, but isn't this only checking the first element of the Java path vec? Did you instead mean to iterate, map, and take the first one that was valid?

Reply:
Yeah, this grabs the first found compatible version. If length is higher than 1, the rest will be ignored.

Reply:
Oh I understand. So once youve gotten the valid paths, you only care about getting the first one, because the rest are redundant.

Resolving.